### DISCUSSION TITLE
Can segment carry more information?

### DISCUSSION THREAD

**Discussion by CWgo2000** [apache/skywalking#12360] [General]
I used to use Zipkin, which can carry information such as client ip,I got the client related information from the parsentSegmentId in refs, but if the client does not have the skywalking agent installed, or the service is not Java, the client ip cannot be obtained.How to handle this situation, or if the segment carries the client IP from the beginning, can Skywalking be supported? What should I do?

**Comment by wu-sheng** [comment]
No, we don't support this case.
Tracing aims to do end to end, you could tag IP in span if you want, but it doesn't help on our major purpose, which is 100% sampling based topology build and accurate metrics.

We have supported Zipkin as a alternative server, but as how Zipkin is designed, no topo and metric are there.

We are different systems, and have different landscapes.

**Reply by CWgo2000** [reply]
Thank you for your reply. Skywalking has a better monitoring UI, and I will try other ways to slove this issue, such as agent plugins.

**Reply by wu-sheng** [reply]
Like I said, if you want to collect peer IP, collect them.
SkyWalking doesn't collect them mandatorily.

**Comment by wu-sheng** [comment]
If Zipkin is what you want, why you bother asking us to do that?
